test/foundry/deploy/15_holograph_registry_tests.t.sol

```diff
@@ -0,0 +1,802 @@
+        //bytes memory  encodeSignature = abi.encodeWithSignature('isHolographedHashDeployed(bytes32)', sampleERC721Hash);
+        //mockExternalCall.callExternalFn(address(registry), encodeSignature);        
+    }    
+
+/*
+    getHolographedHashAddress
+*/
+
+/**
+ * @notice Test the validity of the holographed contracts hash map in the registry.
+ * @dev This test checks if the registry correctly maps the hash of sampleErc721Holographer
+ * to the address of the sampleErc721Holographer contract.
+ * It verifies that the function returns the correct address for a given holographed contract hash.
+ * Refers to the hardhat test with the description 'Should return valid _holographedContractsHashMap'
+ */
+    function testReturnValid_holographedContractsHashMap() public {
+        vm.skip(true);
+        //address add = registry.getHolographedHashAddress(sampleErc721Hash);
+        //assertEq(add, address(sampleErc721Holographer));
+    }    
+
+/**
+ * @notice Test the return of the zero address for an invalid holographed contract hash.
+ * @dev This test checks if the registry correctly returns the zero address
+ * when queried for the address of an invalid holographed contract hash.
+ * It verifies that the function returns the zero address for an invalid hash.
+ * Refers to the hardhat test with the description 'should return 0x0 for invalid hash'
+ */
+    function testReturn0x0ForInvalidHash() public {
+        address add = registry.getHolographedHashAddress(contractHash);
+        assertEq(add, zeroAddress);
+    }    
+
+/**
+ * @notice Test allowing an external contract to call the getHolographedHashAddress function.
+ * @dev This test verifies that an external contract can successfully call the getHolographedHashAddress function
+ * of the registry contract with the hash of sampleERC721Hash as a parameter.
+ * Refers to the hardhat test with the description 'Should allow external contract to call fn'
+ */
+    function testAllowExternalContractToCalGetHolographedHashAddress() public {
+        vm.skip(true);
+        //bytes memory  encodeSignature = abi.encodeWithSignature('getHolographedHashAddress(bytes32)', sampleERC721Hash);
+        //mockExternalCall.callExternalFn(address(registry), encodeSignature);         
+    }
+
+/*
+    setReservedContractTypeAddress
+*/
+
+/**
+ * @notice Test allowing an admin to set a contract type address in the registry.
+ * @dev This test verifies that an admin can successfully set a reserved contract type address
+ * in the registry for the contract type 'HolographERC721'.
+ * It simulates the admin (deployer) calling the setReservedContractTypeAddress function with the contract type hash and a boolean value.
+ * Refers to the hardhat test with the description 'should allow admin to set contract type address'
+ */
+    function testAllowAdminToSetContractTypeAddress() public {
+        vm.prank(deployer);
+        registry.setReservedContractTypeAddress(contractHash, true);
```

Comment:
do these functions where we expect the tx not to reverse, do they broadcast an event? if so, we could evaluate the broadcasting of that event.